From e75bde233b9fe6b1a7d24cc63fa9baf9e89de895 Mon Sep 17 00:00:00 2001
From: Dominik Haumann <dhaumann@kde.org>
Date: Tue, 20 Mar 2018 19:51:57 +0100
Subject: Revert Fix: View jumps when Scroll past end of document is enabled

Unfortunately, this introduced a regression that scrolling down in
dynamically wrapped lines did not move the view at all. We need
to find a better fix, and introduce a unit test.

CCBUG: 306745
BUG: 391838
FIXED-IN: 5.45
---
 autotests/src/kateview_test.cpp | 2 ++
 src/view/kateviewinternal.cpp   | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/autotests/src/kateview_test.cpp b/autotests/src/kateview_test.cpp
index ba9cd21..d9b5672 100644
--- a/autotests/src/kateview_test.cpp
+++ b/autotests/src/kateview_test.cpp
@@ -293,6 +293,7 @@ void KateViewTest::testKillline()
 
 void KateViewTest::testScrollPastEndOfDocument()
 {
+#if 0 // bug still exists, see bug 306745
     KTextEditor::DocumentPrivate doc;
     doc.setText(QStringLiteral("0000000000\n"
                                "1111111111\n"
@@ -322,6 +323,7 @@ void KateViewTest::testScrollPastEndOfDocument()
     QCOMPARE(view->cursorPosition(), KTextEditor::Cursor(4, 5));
     // verify, that only lines 3333333333 and 4444444444 are still visible.
     QCOMPARE(view->firstDisplayedLineInternal(KTextEditor::View::RealLine), 3);
+#endif
 }
 
 void KateViewTest::testFoldFirstLine()
diff --git a/src/view/kateviewinternal.cpp b/src/view/kateviewinternal.cpp
index dc88c94..c56bcff 100644
--- a/src/view/kateviewinternal.cpp
+++ b/src/view/kateviewinternal.cpp
@@ -709,7 +709,7 @@ void KateViewInternal::makeVisible(const KTextEditor::Cursor &c, int endCol, boo
     } else if (center && (c < startPos() || c > endPos())) {
         KTextEditor::Cursor scroll = viewLineOffset(c, -int(linesDisplayed()) / 2);
         scrollPos(scroll, false, calledExternally);
-    } else if (c.line() > viewLineOffset(startPos(), linesDisplayed() - m_minLinesVisible - 1).line()) {
+    } else if (c > viewLineOffset(startPos(), linesDisplayed() - m_minLinesVisible - 1)) {
         KTextEditor::Cursor scroll = viewLineOffset(c, -(linesDisplayed() - m_minLinesVisible - 1));
         scrollPos(scroll, false, calledExternally);
     } else if (c < viewLineOffset(startPos(), m_minLinesVisible)) {
-- 
cgit v0.11.2

